x-pack/plugins/cases/public/containers/use_messages_storage.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo } from 'react';
+import { Storage } from '../../../../../src/plugins/kibana_utils/public';
+
+export interface UseMessagesStorage {
+  getMessages: (plugin: string) => string[];
+  addMessage: (plugin: string, id: string) => void;
+  removeMessage: (plugin: string, id: string) => void;
+  clearAllMessages: (plugin: string) => void;
+  hasMessage: (plugin: string, id: string) => boolean;
+}
+
+// TODO: Removed const { storage } = useKibana().services; in favor of using the util
```

Comment:
whats the to do? is this kosher?

Reply:
See earlier comment: https://github.com/elastic/kibana/pull/94840#discussion_r597004248